controller/organization.go

```diff
@@ -0,0 +1,161 @@
+package controller
+
+import (
+	"github.com/fabric8-services/fabric8-auth/account"
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/application"
+	"github.com/fabric8-services/fabric8-auth/authorization/resource"
+	"github.com/fabric8-services/fabric8-auth/authorization/role"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/login"
+	"github.com/fabric8-services/fabric8-auth/token"
+	"github.com/goadesign/goa"
+	"github.com/satori/go.uuid"
+	"strings"
+)
+
+const OrganizationOwnerRole = "owner"
+
+// OrganizationController implements the organization resource.
+type OrganizationController struct {
+	*goa.Controller
+	db           application.DB
+	TokenManager token.Manager
+}
+
+// NewOrganizationController creates an organization controller.
+func NewOrganizationController(service *goa.Service, db application.DB) *OrganizationController {
+	return &OrganizationController{Controller: service.NewController("OrganizationController"), db: db}
+}
+
+// Create runs the create action.
+func (c *OrganizationController) Create(ctx *app.CreateOrganizationContext) error {
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+
+	if len(strings.TrimSpace(*ctx.Payload.Name)) == 0 {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest("Organization name cannot be empty"))
+	}
+
+	var organizationId uuid.UUID
+
+	err = application.Transactional(c.db, func(appl application.Application) error {
+
+		// Lookup the identity for the current user
+		userIdentity, err := appl.Identities().Load(ctx, *currentUser)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewInternalErrorFromString(ctx, "Error looking up current user"))
```

Comment:
`appl.Identities().Load(ctx, *currentUser)` also returns an error if the user doesn't exist in the db, and in that case it is should be reported by the API as an unauthorized error  
```
return errors.NewUnauthorizedError(fmt.Sprintf("auth token contains id %s of unknown Identity\n", *id))
```

Reply:
Thanks, I've updated this.